src/Widgets/calendar/CalendarView.vala

```diff
@@ -241,8 +249,11 @@ public class DateTime.Widgets.CalendarView : Gtk.Grid {
     }
 
     // TODO: As far as maya supports it use the Dbus Activation feature to run the calendar-app.
-    public void show_date_in_maya (GLib.DateTime date) {
-        var command = "io.elementary.calendar --show-day %s".printf (date.format ("%F"));
+    private void open_maya_with_options (GLib.DateTime? day_to_show, bool add_event = false) {
```

Comment:
Expanded the previous `show_date_in_maya` method to also allow for displaying the new event dialog. Public API remains the same and available.